{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from confluent_kafka import Producer\n",
    "import json\n",
    "\n",
    "KAFKA_TOPIC = 'latest_events'\n",
    "KAFKA_BOOTSTRAP_SERVERS = 'broker:9092'\n",
    "#REQUEST_URL = \"https://stream.wikimedia.org/v2/stream/recentchange\"\n",
    "REQUEST_URL = \"http://fact_api/fact\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def delivery_callback(err, msg):\n",
    "    if err:\n",
    "        print('%% Message failed delivery: %s' % err)\n",
    "    else:\n",
    "        print('%% Message delivered to %s [%d]' % (msg.topic(), msg.partition()))\n",
    "\n",
    "def main():\n",
    "    # Kafka producer configuration\n",
    "    conf = {'bootstrap.servers': KAFKA_BOOTSTRAP_SERVERS}\n",
    "\n",
    "    # Create Kafka producer instance\n",
    "    producer = Producer(conf)\n",
    "\n",
    "    # Read the Wikimedia stream and produce messages to Kafka\n",
    "    response = requests.get(REQUEST_URL).json()\n",
    "    \n",
    "    \n",
    "    for adict in response:\n",
    "        if adict:\n",
    "            message = json.dumps(adict)            \n",
    "            \n",
    "            # Produce message to Kafka\n",
    "            producer.produce(KAFKA_TOPIC, message, callback=delivery_callback)\n",
    "\n",
    "            # Flush messages to Kafka to ensure they are sent immediately\n",
    "            producer.flush()\n",
    "\n",
    "    # Close Kafka producer\n",
    "    producer.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "%3|1710510477.981|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 161ms in state CONNECT)\n"
     ]
    },
    {
     "ename": "ConnectionError",
     "evalue": "HTTPConnectionPool(host='fact_api', port=80): Max retries exceeded with url: /fact (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1170f08d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mgaierror\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connection.py:174\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 174\u001b[0m     conn \u001b[38;5;241m=\u001b[39m connection\u001b[38;5;241m.\u001b[39mcreate_connection(\n\u001b[1;32m    175\u001b[0m         (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dns_host, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mport), \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtimeout, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mextra_kw\n\u001b[1;32m    176\u001b[0m     )\n\u001b[1;32m    178\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketTimeout:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/util/connection.py:72\u001b[0m, in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[1;32m     68\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m six\u001b[38;5;241m.\u001b[39mraise_from(\n\u001b[1;32m     69\u001b[0m         LocationParseError(\u001b[38;5;124mu\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m, label empty or too long\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m host), \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m     70\u001b[0m     )\n\u001b[0;32m---> 72\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m res \u001b[38;5;129;01min\u001b[39;00m socket\u001b[38;5;241m.\u001b[39mgetaddrinfo(host, port, family, socket\u001b[38;5;241m.\u001b[39mSOCK_STREAM):\n\u001b[1;32m     73\u001b[0m     af, socktype, proto, canonname, sa \u001b[38;5;241m=\u001b[39m res\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/socket.py:962\u001b[0m, in \u001b[0;36mgetaddrinfo\u001b[0;34m(host, port, family, type, proto, flags)\u001b[0m\n\u001b[1;32m    961\u001b[0m addrlist \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m--> 962\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m res \u001b[38;5;129;01min\u001b[39;00m _socket\u001b[38;5;241m.\u001b[39mgetaddrinfo(host, port, family, \u001b[38;5;28mtype\u001b[39m, proto, flags):\n\u001b[1;32m    963\u001b[0m     af, socktype, proto, canonname, sa \u001b[38;5;241m=\u001b[39m res\n",
      "\u001b[0;31mgaierror\u001b[0m: [Errno 8] nodename nor servname provided, or not known",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mNewConnectionError\u001b[0m                        Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connectionpool.py:714\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    713\u001b[0m \u001b[38;5;66;03m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[0;32m--> 714\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_make_request(\n\u001b[1;32m    715\u001b[0m     conn,\n\u001b[1;32m    716\u001b[0m     method,\n\u001b[1;32m    717\u001b[0m     url,\n\u001b[1;32m    718\u001b[0m     timeout\u001b[38;5;241m=\u001b[39mtimeout_obj,\n\u001b[1;32m    719\u001b[0m     body\u001b[38;5;241m=\u001b[39mbody,\n\u001b[1;32m    720\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[1;32m    721\u001b[0m     chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[1;32m    722\u001b[0m )\n\u001b[1;32m    724\u001b[0m \u001b[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[1;32m    725\u001b[0m \u001b[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[1;32m    726\u001b[0m \u001b[38;5;66;03m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[1;32m    727\u001b[0m \u001b[38;5;66;03m# mess.\u001b[39;00m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connectionpool.py:415\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    414\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 415\u001b[0m         conn\u001b[38;5;241m.\u001b[39mrequest(method, url, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mhttplib_request_kw)\n\u001b[1;32m    417\u001b[0m \u001b[38;5;66;03m# We are swallowing BrokenPipeError (errno.EPIPE) since the server is\u001b[39;00m\n\u001b[1;32m    418\u001b[0m \u001b[38;5;66;03m# legitimately able to close the connection after sending a valid response.\u001b[39;00m\n\u001b[1;32m    419\u001b[0m \u001b[38;5;66;03m# With this behaviour, the received response is still readable.\u001b[39;00m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connection.py:244\u001b[0m, in \u001b[0;36mHTTPConnection.request\u001b[0;34m(self, method, url, body, headers)\u001b[0m\n\u001b[1;32m    243\u001b[0m     headers[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUser-Agent\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m _get_default_user_agent()\n\u001b[0;32m--> 244\u001b[0m \u001b[38;5;28msuper\u001b[39m(HTTPConnection, \u001b[38;5;28mself\u001b[39m)\u001b[38;5;241m.\u001b[39mrequest(method, url, body\u001b[38;5;241m=\u001b[39mbody, headers\u001b[38;5;241m=\u001b[39mheaders)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/http/client.py:1283\u001b[0m, in \u001b[0;36mHTTPConnection.request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1282\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Send a complete request to the server.\"\"\"\u001b[39;00m\n\u001b[0;32m-> 1283\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_send_request(method, url, body, headers, encode_chunked)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/http/client.py:1329\u001b[0m, in \u001b[0;36mHTTPConnection._send_request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1328\u001b[0m     body \u001b[38;5;241m=\u001b[39m _encode(body, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbody\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m-> 1329\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mendheaders(body, encode_chunked\u001b[38;5;241m=\u001b[39mencode_chunked)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/http/client.py:1278\u001b[0m, in \u001b[0;36mHTTPConnection.endheaders\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1277\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CannotSendHeader()\n\u001b[0;32m-> 1278\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_send_output(message_body, encode_chunked\u001b[38;5;241m=\u001b[39mencode_chunked)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/http/client.py:1038\u001b[0m, in \u001b[0;36mHTTPConnection._send_output\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1037\u001b[0m \u001b[38;5;28;01mdel\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_buffer[:]\n\u001b[0;32m-> 1038\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(msg)\n\u001b[1;32m   1040\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m message_body \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1041\u001b[0m \n\u001b[1;32m   1042\u001b[0m     \u001b[38;5;66;03m# create a consistent interface to message_body\u001b[39;00m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/http/client.py:976\u001b[0m, in \u001b[0;36mHTTPConnection.send\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    975\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mauto_open:\n\u001b[0;32m--> 976\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mconnect()\n\u001b[1;32m    977\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connection.py:205\u001b[0m, in \u001b[0;36mHTTPConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    204\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mconnect\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 205\u001b[0m     conn \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_new_conn()\n\u001b[1;32m    206\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_conn(conn)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connection.py:186\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m--> 186\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m NewConnectionError(\n\u001b[1;32m    187\u001b[0m         \u001b[38;5;28mself\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFailed to establish a new connection: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m e\n\u001b[1;32m    188\u001b[0m     )\n\u001b[1;32m    190\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m conn\n",
      "\u001b[0;31mNewConnectionError\u001b[0m: <urllib3.connection.HTTPConnection object at 0x1170f08d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/adapters.py:487\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    486\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 487\u001b[0m     resp \u001b[38;5;241m=\u001b[39m conn\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[1;32m    488\u001b[0m         method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[1;32m    489\u001b[0m         url\u001b[38;5;241m=\u001b[39murl,\n\u001b[1;32m    490\u001b[0m         body\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mbody,\n\u001b[1;32m    491\u001b[0m         headers\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[1;32m    492\u001b[0m         redirect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    493\u001b[0m         assert_same_host\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    494\u001b[0m         preload_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    495\u001b[0m         decode_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[1;32m    496\u001b[0m         retries\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_retries,\n\u001b[1;32m    497\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[1;32m    498\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[1;32m    499\u001b[0m     )\n\u001b[1;32m    501\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/connectionpool.py:798\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    796\u001b[0m     e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, e)\n\u001b[0;32m--> 798\u001b[0m retries \u001b[38;5;241m=\u001b[39m retries\u001b[38;5;241m.\u001b[39mincrement(\n\u001b[1;32m    799\u001b[0m     method, url, error\u001b[38;5;241m=\u001b[39me, _pool\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m, _stacktrace\u001b[38;5;241m=\u001b[39msys\u001b[38;5;241m.\u001b[39mexc_info()[\u001b[38;5;241m2\u001b[39m]\n\u001b[1;32m    800\u001b[0m )\n\u001b[1;32m    801\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/urllib3/util/retry.py:592\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m new_retry\u001b[38;5;241m.\u001b[39mis_exhausted():\n\u001b[0;32m--> 592\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause))\n\u001b[1;32m    594\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPConnectionPool(host='fact_api', port=80): Max retries exceeded with url: /fact (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1170f08d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConnectionError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m main()\n",
      "Cell \u001b[0;32mIn[3], line 15\u001b[0m, in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m     12\u001b[0m producer \u001b[38;5;241m=\u001b[39m Producer(conf)\n\u001b[1;32m     14\u001b[0m \u001b[38;5;66;03m# Read the Wikimedia stream and produce messages to Kafka\u001b[39;00m\n\u001b[0;32m---> 15\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mget(REQUEST_URL)\u001b[38;5;241m.\u001b[39mjson()\n\u001b[1;32m     18\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m adict \u001b[38;5;129;01min\u001b[39;00m response:\n\u001b[1;32m     19\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m adict:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/api.py:73\u001b[0m, in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[1;32m     64\u001b[0m \n\u001b[1;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m request(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mget\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, params\u001b[38;5;241m=\u001b[39mparams, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m session\u001b[38;5;241m.\u001b[39mrequest(method\u001b[38;5;241m=\u001b[39mmethod, url\u001b[38;5;241m=\u001b[39murl, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/sessions.py:587\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    582\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    583\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    584\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    585\u001b[0m }\n\u001b[1;32m    586\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 587\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[1;32m    589\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/sessions.py:701\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    698\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    700\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 701\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[1;32m    703\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    704\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[0;32m/opt/anaconda3/envs/dev311/lib/python3.11/site-packages/requests/adapters.py:520\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    516\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    517\u001b[0m         \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[1;32m    518\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[0;32m--> 520\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    522\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m ClosedPoolError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    523\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "\u001b[0;31mConnectionError\u001b[0m: HTTPConnectionPool(host='fact_api', port=80): Max retries exceeded with url: /fact (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x1170f08d0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "%3|1710510479.825|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 1 identical error(s) suppressed)\n",
      "%3|1710510509.907|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510539.993|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510570.082|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510600.167|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510630.242|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510660.324|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510690.404|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510720.505|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510750.595|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510781.683|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710510811.762|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510841.845|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510871.922|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510903.019|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710510934.121|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710510964.203|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710510994.290|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511025.371|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511055.448|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511085.537|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511116.642|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511146.737|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511176.832|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511206.931|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511237.017|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511267.101|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511297.184|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511327.268|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511357.352|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511387.429|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511417.513|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511447.595|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511477.682|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511507.754|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511538.833|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511568.907|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 7ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511598.984|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511630.073|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511660.159|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511691.262|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511721.353|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511751.444|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 6ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511781.530|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511811.611|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511842.685|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710511872.772|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511902.848|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511932.940|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511963.017|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710511993.094|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512023.196|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512053.284|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512083.375|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512113.461|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 9ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512144.527|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512175.616|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512205.699|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512235.784|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512265.883|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 16ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512296.955|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512328.052|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512358.139|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512388.230|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 8ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512418.301|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512448.386|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512478.471|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512508.554|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 7ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512538.635|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512568.702|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512599.794|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512629.875|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512659.970|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 8ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512691.049|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710512721.134|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512751.213|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512781.282|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512811.370|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512841.463|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512871.678|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512901.908|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512932.116|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512962.322|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710512992.413|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513023.506|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513054.588|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513084.673|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513115.764|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513145.844|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513175.919|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513207.009|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513237.091|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513267.190|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513297.266|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513327.365|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513357.444|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513387.532|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513417.625|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513447.699|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513477.784|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513507.870|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513537.958|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 7ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513568.041|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513598.237|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513629.343|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513659.431|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513689.524|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513719.604|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513749.685|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513779.776|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513809.862|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513839.939|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513870.026|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513900.114|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710513931.207|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513962.299|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710513992.363|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514022.438|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514052.523|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514083.613|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514113.697|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514143.784|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514173.878|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514203.966|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514234.049|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514264.145|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514295.216|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514325.300|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514355.380|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514385.462|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514415.544|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514446.616|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514476.710|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514506.793|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514537.874|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514567.971|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514599.047|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514629.130|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514660.218|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514691.306|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710514721.383|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514751.454|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514781.536|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514811.616|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514841.706|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514871.798|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514901.875|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514931.965|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514962.053|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710514993.139|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710515023.226|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515053.306|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515083.406|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515113.486|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515143.561|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515173.640|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515204.740|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710515235.828|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710515265.916|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 6ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515297.019|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 16 identical error(s) suppressed)\n",
      "%3|1710515327.102|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515357.189|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515387.270|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515417.353|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515447.437|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 5ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515477.507|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 2ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515507.591|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515537.674|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515567.761|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515597.879|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515628.102|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515658.294|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 4ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515688.520|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515718.714|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515748.948|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n",
      "%3|1710515779.156|FAIL|rdkafka#producer-1| [thrd:broker:9092/bootstrap]: broker:9092/bootstrap: Failed to resolve 'broker:9092': nodename nor servname provided, or not known (after 3ms in state CONNECT, 15 identical error(s) suppressed)\n"
     ]
    }
   ],
   "source": [
    "main()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dev311",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
